### ISSUE TITLE
Canu is hanging at step ovlStore

### ISSUE THREAD

**Issue by geycu** [marbl/canu#252]
Hello,

We are using Canu for a genome assembly. Unfortunately, we are thinking that Canu is hanging and does not produce an output. Therefore, we are contacting you to see whether you can provide us some help. We have summarized some details about the current “step” Canu is performing as well as on the setup of our cluster.

**Genome to be assembled and data:**
- 1 gb genome
- 50 million PacBio reads
- ~50x coverage

**Current situation of Canu**
We are performing with Canu the “correct” step. Here, we are observing that a specific step now took more than 11 days without any significant process –or at least, we are not seeing any progress. In particular, Canu stays in the step “ovS" as listed as job name by Slurm. Here on each of the compute nodes, the script “correction/correct.ovlStore.BUILDING/scripts/2-sort.sh” is being executed which – in turn – is running the command

/mnt/lustre/sw/software/canu/c2e0681-foss-2014b/bin/ovStoreSorter -deletelate -M 205 -O /mnt/lustre/ec2-user/xxxx/ts/correct_full/correct_output/correction/correct.ovlStore.BUILDING -G /mnt/lustre/ec2-user/xxxx/ts/correct_full/correct_output/correction/correct.gkpStore -F 9 -job <<ID>> 17

where <<ID>> ranges from 1 to 9. This command keeps one core busy i.e., 100% CPU load is observed for this process.

**We have initiated the Canu run with the following parameters**

srun canu -correct \
-p correct_canola_full \
-d ${OUTDIR} \
genomeSize=1g \
useGrid=1 \
merylThreads=32 \
merylMemory=230 \
cormhapThreads=32 \
cormhapMemory=230 \
corThreads=32 \
corMemory=230 \
obtovlThreads=32 \
obtovlMemory=230 \
utgovlThreads=32 \
utgovlMemory=230 \
redThreads=32 \
redMemory=230 \
oeaThreads=32 \
oeaMemory=230 \
cnsThreads=32 \
cnsMemory=230 \
ovsMemory=230 \
ovsThreads=32 \
ovsMethod=parallel \
batThreads=32 \
batMemory=230 \
-pacbio-raw ${INPUT}

**Setup of the cluster**
We are using a cluster consisting of 10 machines each equipped with 32 cores (Intel(R) Xeon(R) CPU E5-2670 v2 @ 2.50GHz), ~245GB of RAM and a shared, parallel file system of 10TB where ~50% are used. The jobs are handled by slurm as a batch management system. We compiled Canu by ourselves using the git version “c2e0681ba98711641e0a29e28f0dbdd071c2ee41” of Canu. Therefore, we have used the GCC 4.8.3, Perl 5.20.0 and Java 1.8.0_65.

We'd really appreciate if you could help us to figure this out.

Thanks a lot in advance!!

**Comment by brianwalenz** [comment]
Very nice bug report.

All the threads and memory options are not needed, and in this case, doing the wrong thing for you.  Canu will auto-detect the hosts on your grid, and using genomeSize, will set memory and threads appropriately.

In this case, the overlap store building - essentially sorting all the overlaps - split the task into 9 jobs, each very very large.  The 'usual' size for these jobs is 4-8gb.

Your choice is to let the current jobs finish (I have no way of estimating how long this will take) or to kill what you have, remove the memory and threads options, remove the correct.ovlStore.BUILDING directory and files and restart.  If any of the ovStoreSorter jobs have finished, the input needed for restart will have been deleted.  This can be recomputed (automatically) with a little bit of additional cleanup.

**Comment by geycu** [comment]
Thank you very much for the prompt reply!

Following your suggestion, we have restarted Canu without providing any memory/thread parameter, i.e., the job file reads as follows

srun canu -correct \
-p correct_canola_full \
-d ${OUTDIR} \
genomeSize=1g \
useGrid=1 \
ovsMethod=parallel \
-pacbio-raw ${INPUT}

The run is finished. Unfortunately, no fasta file has been produced as an output of the correct step. In the logs of the scheduler, we observe, that all ovS_correct* jobs have been completed but the job named “canu_correct_canola_full” did not finished successful. 

We are observing that a directory “correction/correct_canola_full.ovlStore” has been created including an empty file named “2-sorter.success”. However, there is no directory “correction/2-correction” as observed in the previous run.

Can you help us here what is missing?

Please do not hesitate to ask if you have more questions.

Thanks a lot in advance!

**Comment by geycu** [comment]
Hello,

I am wondering whether you've had chance to look at this and provide us any advice? 

Thanks!

**Comment by skoren** [comment]
The next step after sorting builds an index and continues on with the assembly which will create the 2-correction folder. What is the output in the last canu.[0-9]\*.out files in canu-scripts? That should provide more information on what happened with the run.

**Comment by geycu** [comment]
The last step is canu.07:

-- overlap store sorter attempt 0 begins with 0 finished, and 239 to compute.
----------------------------------------
-- Starting command on Mon Sep 19 14:48:59 2016 with 8554.2 GB free disk space

      sbatch \
        --mem=5g --cpus-per-task=1 \
        -D `pwd` -J "ovS_correct_wheat_full" \
        -a 1-239 \
        -o /mnt/lustre/ec2-user/wheat/correct_5million/correct_output/correction/correct_wheat_full.ovlStore.BUILDING/logs/2-sort.%A_%a.out \
        /mnt/lustre/ec2-user/wheat/correct_5million/correct_output/correction/correct_wheat_full.ovlStore.BUILDING/scripts/2-sort.sh

Submitted batch job 743

-- Finished on Mon Sep 19 14:48:59 2016 (lickety-split) with 8554.2 GB free disk space

-- Starting command on Mon Sep 19 15:22:56 2016 with 8281.6 GB free disk space

    /mnt/lustre/sw/software/canu/c2e0681-foss-2014b/bin/ovStoreStats \
     -G /mnt/lustre/ec2-user/wheat/correct_5million/correct_output/correction/correct_wheat_full.gkpStore \
     -O /mnt/lustre/ec2-user/wheat/correct_5million/correct_output/correction/correct_wheat_full.ovlStore \
     -o /mnt/lustre/ec2-user/wheat/correct_5million/correct_output/correction/correct_wheat_full.ovlStore \
     > /mnt/lustre/ec2-user/wheat/correct_5million/correct_output/correction/correct_wheat_full.ovlStore.summary.err 2>&1
slurmstepd: error: Job 744 exceeded memory limit (4215344 > 4194304), being killed
slurmstepd: error: Exceeded job memory limit
slurmstepd: error: *** JOB 744 ON cls-n00 CANCELLED AT 2016-09-19T16:14:46 ***

Is it possible to solve this by setting ovsMemory=200 in the command? 

Thanks!

**Comment by skoren** [comment]
That is very strange output, is that all in one file? It looks like it is submitting the sorting jobs after which it should submit itself and immediately quit. However, the next output you list it is already running overlap store summary which should require an intermediate indexing step. Is it possible this file was edited/truncated since Canu ran?

Regarding the memory limit, the overlap stats are optional so you can re-run the Canu command and it will resume at the next step. You can get around the error by increasing the memory given to the Canu executive script with ``gridOptionsExecutive=--mem=30g``` replacing 30 with the approximate size of the correction/asm.gkpStore/blobs file. The issue in 1.3 is that Canu used memory mapped file access and depending on OS and scheduler the virtual memory in the file would either count against your process or not, causing the job to fail even though the virtual memory could be paged out at any time (it is already on disk). We've also changed how Canu processes the read data so this shouldn't happen in the latest code (only 1.3).

**Comment by geycu** [comment]
I use the suggested memory setting --mem=30g, and it worked perfect!!! Thanks!

Now the correction and trim steps are finished, but canu -assemble failed at canu.08.out:

-- This is canu parallel iteration #3, out of a maximum of 2 attempts.
--
-- Final error rates before starting pipeline:
--
--   genomeSize          -- 1000000000
--   errorRate           -- 0.025
--
--   corOvlErrorRate     -- 0.075
--   obtOvlErrorRate     -- 0.075
--   utgOvlErrorRate     -- 0.075
--
--   obtErrorRate        -- 0.075
--
--   utgGraphErrorRate   -- 0.05
--   utgBubbleErrorRate  -- 0.0625
--   utgMergeErrorRate   -- 0.0375
--   utgRepeatErrorRate  -- 0.05
--
--   cnsErrorRate        -- 0.0625
--
--
-- BEGIN ASSEMBLY
--
--
-- 32 overlap error adjustment jobs failed:
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0001.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0002.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0003.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0004.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0005.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0006.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0007.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0008.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0009.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0010.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0011.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0012.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0013.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0014.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0015.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0016.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0017.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0018.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0019.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0020.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0021.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0022.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0023.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0024.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0025.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0026.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0027.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0028.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0029.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0030.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0031.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output1/unitigging/3-overlapErrorAdjustment/0032.oea FAILED.
--
================================================================================
Don't panic, but a mostly harmless error occurred and canu failed.

canu failed with 'failed to adjust overlap error rates.  Made 2 attempts, jobs still failed'.

(END)


Any suggestion? Do I need to adjust any other error rates?

Thanks!

**Comment by geycu** [comment]
Hi,

I am wondering whether you've had chance to look at this?  

Thanks!

**Comment by brianwalenz** [comment]
Look in the `unitigging/3-overlapErrorAdjustment` directory for any \*.out or \*.err files.  They're probably named something similar to `oea.1.out`.  I will guess it is (again) being killed due to memory.  Submitting them by hand - look in canu.07.out for the command - while increasing the slurm memory request will be the easiest way to get around this.  When they're finished running, restart canu as before.

**Comment by geycu** [comment]
I increased the memory from --mem=30g to --mem=100g, which should be sufficient. However it failed again at canu.07.out/canu.08.out:

-- 32 overlap error adjustment jobs failed:
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0001.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0002.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0003.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0004.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0005.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0006.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0007.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0008.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0009.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0010.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0011.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0012.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0013.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0014.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0015.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0016.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0017.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0018.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0019.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0020.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0021.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0022.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0023.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0024.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0025.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0026.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0027.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0028.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0029.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0030.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0031.oea FAILED.
--   job /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/0032.oea FAILED.
--
-- overlap error adjustment attempt 2 begins with 0 finished, and 32 to compute.
----------------------------------------
-- Starting command on Wed Oct 12 05:09:39 2016 with 2389 GB free disk space

      sbatch \
        --mem=2g --cpus-per-task=1 \
        -D `pwd` -J "oea_asm_canola_full" \
        -a 1-32 \
        -o /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/oea.%A_%a.out \
        /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/oea.sh

Submitted batch job 4106

-- Finished on Wed Oct 12 05:09:39 2016 (lickety-split) with 2389 GB free disk space
----------------------------------------
----------------------------------------
-- Starting command on Wed Oct 12 05:09:39 2016 with 2389 GB free disk space

    sbatch \
      --mem=4g \
      --cpus-per-task=1  \
      --mem=100g \
      --depend=afterany:4106 \
      -D `pwd` \
      -J "canu_asm_canola_full" \
      -o /mnt/lustre/ec2-user/canola/correct_full/asm_output2/canu-scripts/canu.08.out /mnt/lustre/ec2-user/canola/correct_full/asm_output2/canu-scripts/canu.08.sh
Submitted batch job 4107

-- Finished on Wed Oct 12 05:09:39 2016 (lickety-split) with 2389 GB free disk space
----------------------------------------
(END)

**Comment by geycu** [comment]
Hello,

I am wondering whether you had chance to look at this. 

I checked oea.*.out and all the jobs crashed due to memory again:

slurmstepd: error: Job 4007 exceeded memory limit (2350004 > 2097152), being killed
slurmstepd: error: Exceeded job memory limit

When I checked the command canu.07.out, I see the run only used 2g:
      sbatch \
        --mem=2g --cpus-per-task=1 \
        -D `pwd` -J "oea_asm_canola_full" \
        -a 1-32 \
        -o /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/oea.%A_%a.out \
        /mnt/lustre/ec2-user/canola/correct_full/asm_output2/unitigging/3-overlapErrorAdjustment/oea.sh

Is there any way I can pass my memory setting (in my case --mem=30g) to this step? Or using --mem=30g for other steps, but specify another memory setting for this step?

Thanks!!

**Comment by brianwalenz** [comment]
I think we've fixed this already.  Your version seems to be from early March.  I don't recommend updating the code in the middle of an assembly though.

If you can afford to recompute overlaps, by all means, do update!  There are significant improvements in contig construction.  You can reuse the trimmed reads and compute just the assembly:
`canu ... -assemble -pacbio-corrected asm_output2/*.trimmedReads.fasta.gz`

If you can't afford to, or don't want to, I think the easiest way forward will be to submit these jobs by hand, adjusting --mem as needed.  Just submit the `sbatch` you've posted, wait for it to finish, and restart canu.  It'll check that all jobs finished and resume the assembly.

**Comment by geycu** [comment]
Thanks!

We downloaded and updated the version and re-run from the very beginning, using the command:

srun canu -correct \
-p correct_full \
-d ${OUTDIR} \
gridOptionsExecutive="--mem=30g" \
genomeSize=1g \
useGrid=1 \
ovsMethod=parallel \
gnuplotTested=true \
-pacbio-raw ${INPUT}

The errors showed at the step mhap precomputer (canu.02.out):
-- 37 mhap precompute jobs failed:
..........
-- mhap precompute attempt 2 begins with 14 finished, and 37 to compute.
----------------------------------------
-- Starting command on Fri Oct 21 12:35:13 2016 with 9966.315 GB free disk space

      sbatch \
        --mem-per-cpu=2g --cpus-per-task=16 \
        -D `pwd` -J "cormhap_correct_canola_full" \
        -a 2-2 \
        -o /mnt/lustre/data/Canu_runs/Canola/full/correct_output2/correction/1-overlapper/precompute.%A_%a.out \
        /mnt/lustre/data/Canu_runs/Canola/full/correct_output2/correction/1-overlapper/precompute.sh 0

Submitted batch job 4991

-- Finished on Fri Oct 21 12:35:13 2016 (lickety-split) with 9966.315 GB free disk space
----------------------------------------
----------------------------------------
-- Starting command on Fri Oct 21 12:35:13 2016 with 9966.315 GB free disk space

      sbatch \
        --mem-per-cpu=2g --cpus-per-task=16 \
        -D `pwd` -J "cormhap_correct_canola_full" \
        -a 16-51 \
        -o /mnt/lustre/data/Canu_runs/Canola/full/correct_output2/correction/1-overlapper/precompute.%A_%a.out \
        /mnt/lustre/data/Canu_runs/Canola/full/correct_output2/correction/1-overlapper/precompute.sh 0

Submitted batch job 4992

-- Finished on Fri Oct 21 12:35:13 2016 (lickety-split) with 9966.315 GB free disk space
----------------------------------------
Warning: multiple IDs for job cormhap_correct_canola_full got 2 and should have been 1.
----------------------------------------
-- Starting command on Fri Oct 21 12:35:13 2016 with 9966.315 GB free disk space

    sbatch \
      --mem-per-cpu=4g \
      --cpus-per-task=1  \
      --mem=30g \
      --depend=afterany:4991:4992 \
      -D `pwd` \
      -J "canu_correct_canola_full" \
      -o /mnt/lustre/data/Canu_runs/Canola/full/correct_output2/canu-scripts/canu.03.out /mnt/lustre/data/Canu_runs/Canola/full/correct_output2/canu-scripts/canu.03.sh
Submitted batch job 4993

-- Finished on Fri Oct 21 12:35:13 2016 (lickety-split) with 9966.315 GB free disk space
----------------------------------------

I guess still something wrong with memory?

Thanks!

**Comment by geycu** [comment]
FYI, I also tried cormhapMemory=50 instead of gridOptionsExecutive="--mem=30g", and it still crashed. Any other suggestion?

**Comment by geycu** [comment]
Hi, 

I am wondering whether you've had chance to look at this. Basically I tried different types of memory settings for canu --correct:

- gridOptionsExecutive="--mem=30g" (this worked for the old Canu version)
- cormhapMemory=50
- gridEngineMemoryOption="-l vf=MEMORY" gridOptions="-l h_vmem=50G"

None of these works now, all causes memory issue.

Thanks!

**Comment by skoren** [comment]
Have you tried version 1.4, it may fix this issue as it changed how the reads are stored/accessed to not use memory mapped files.

If it still fails, what is the output in the /correction/1-overlapper/precompute.\*.out

**Comment by brianwalenz** [comment]
Idle.  Open a new issue if needed.
